include/libopencm3/nrf/common/i2c.h

```diff
@@ -0,0 +1,131 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/nrf/memorymap.h>
+#include <libopencm3/nrf/periph.h>
+/**@{*/
+
+/* I2C bus */
+/** @addtogroup i2c_block I2C instances
+ * @{
+ */
+
+#define I2C0            I2C0_BASE
+#define I2C1            I2C1_BASE
+
+/**@}*/
+
+/* Tasks */
+
+#define I2C_TASK_STARTRX(i2c)      MMIO32((i2c) + 0x000)
+#define I2C_TASK_STARTTX(i2c)      MMIO32((i2c) + 0x008)
+#define I2C_TASK_STOP(i2c)         MMIO32((i2c) + 0x014)
+#define I2C_TASK_SUSPEND(i2c)      MMIO32((i2c) + 0x01c)
+#define I2C_TASK_RESUME(i2c)       MMIO32((i2c) + 0x020)
+/* Events */
+
+#define I2C_EVENT_STOPPED(i2c)     MMIO32((i2c) + 0x104)
+#define I2C_EVENT_RXDREADY(i2c)    MMIO32((i2c) + 0x108)
+#define I2C_EVENT_TXDSENT(i2c)     MMIO32((i2c) + 0x11c)
+#define I2C_EVENT_ERROR(i2c)       MMIO32((i2c) + 0x124)
+#define I2C_EVENT_BB(i2c)          MMIO32((i2c) + 0x138)
+#define I2C_EVENT_SUSPENDED(i2c)   MMIO32((i2c) + 0x148)
+/* Registers */
+
+#define I2C_SHORTS(i2c)            MMIO32((i2c) + 0x200)
+#define I2C_INTEN(i2c)             MMIO32((i2c) + 0x300)
+#define I2C_INTENSET(i2c)          MMIO32((i2c) + 0x304)
+#define I2C_INTENCLR(i2c)          MMIO32((i2c) + 0x308)
+#define I2C_ERRORSRC(i2c)          MMIO32((i2c) + 0x4c4)
+#define I2C_ENABLE(i2c)            MMIO32((i2c) + 0x500)
+#define I2C_PSELSCL(i2c)           MMIO32((i2c) + 0x508)
+#define I2C_PSELSDA(i2c)           MMIO32((i2c) + 0x50c)
+#define I2C_RXD(i2c)               MMIO32((i2c) + 0x518)
+#define I2C_TXD(i2c)               MMIO32((i2c) + 0x51c)
+#define I2C_FREQUENCY(i2c)         MMIO32((i2c) + 0x524)
+#define I2C_ADDRESS(i2c)           MMIO32((i2c) + 0x588)
+
+/* Register Contents */
+
+/** @addtogroup i2c_shorts I2C event -> task shortcuts
+ * @{
+ */
+#define I2C_SHORTS_BB_SUSPEND      (1 << 0)
+#define I2C_SHORTS_BB_STOP         (1 << 1)
+
+/**@}*/
+
+/** @addtogroup i2c_interrupts I2C interrupts
+ * @{
+ */
+#define I2C_INTEN_STOPPED          (1 << 1)
+#define I2C_INTEN_RXDREADY         (1 << 2)
+#define I2C_INTEN_TXDSENT          (1 << 7)
+#define I2C_INTEN_ERROR            (1 << 9)
+#define I2C_INTEN_BB               (1 << 14)
+
+/**@}*/
+
+#define I2C_ERRORSRC_OVERRUN       (1 << 0)
+#define I2C_ERRORSRC_ANACK         (1 << 1)
+#define I2C_ERRORSRC_DNACK         (1 << 2)
+
+#define I2C_ENABLE_VALUE           (5)
+
+#define I2C_FREQUENCY_100K         (0x01980000)
+#define I2C_FREQUENCY_250K         (0x04000000)
+#define I2C_FREQUENCY_400K         (0x06680000)
+
+#define I2C_PSEL_OFF               (0xffffffff)
+
+/* Compatibility defines */
+#define i2c0_isr            twi0_isr
+#define i2c1_isr            twi1_isr
```

Comment:
what are these compatible with?  the vector files have "spi0_twi0_isr" so this doesn't look like it would do anything anyway?  Do we even need these?

Reply:
No we don't need them